### PR TITLE
Adding check-branch-name hook.

### PR DESCRIPTION
**Pull request by jmcazaux**
Check that branch-name matches a given regex (passed as argument).

### CODE REVIEW

**Comment by jmcazaux** [comment]
Note to reviewers: this is not meant to replace a "break the CI if branch name does not comply" feature, it will just provide an advance warning to developers if they have created a branch that is not named "the right way".

File: pre_commit_hooks/check-branch-name.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  echo "No regular expression provided to validate the branch name..."
+  echo "Please provide one using the '-r' argument."
+  echo ""
+  echo "Sample 'repo' section of your '.pre-commit-config.yaml' file:"
+  echo "- repo: https://github.com/alma/pre-commit-hooks"
+  echo "  rev: v1.1.0"
+  echo "  hooks:"
+  echo "  - id: check-branch-name"
+  echo "    args:"
+  echo '      - "-r^(ci|docs|feature|fix|hotfix|perf|refactor|test)\/.*$"'
```
Comment (Natim) [inline pre_commit_hooks/check-branch-name.sh]:
```suggestion
  echo '      - "-r^(ci|docs|feature|feat|fix|hotfix|perf|refactor|test)\/.*$"'
```

Comment (jmcazaux) [reply pre_commit_hooks/check-branch-name.sh]: @Natim this is just an example of configuration, so it does not impact what we will actually use.

This stated, `feat` does not currently belong to officially authorized prefixes (according to [The Law](https://github.com/alma/main/blob/main/docs/contributing.md#naming-branches))... ;)

File: .pre-commit-hooks.yaml [hunk 2]
```diff
@@ -10,3 +10,8 @@
   entry: pre_commit_hooks/check-comments.sh
   language: script
   types: [python]
+- id: check-branch-name
+  name: Check the compliance of your branch name
+  description: Checks that the name of the branch you are committing to is compliant with a given regex
+  entry: pre_commit_hooks/check-branch-name.sh
+  language: script
```
Comment (mgu) [inline .pre-commit-hooks.yaml]:
```suggestion
  language: script
  pass_filenames: false
```

File: pre_commit_hooks/check-branch-name.sh (hunk 1 above)
Comment (mgu) [reply pre_commit_hooks/check-branch-name.sh]: maybe should you support the `main` branch in your example

Comment (jmcazaux) [reply pre_commit_hooks/check-branch-name.sh]: @mgu you mean something like:
`^(((ci|docs|feature|fix|hotfix|perf|refactor|test)\/.*)|main)$` 

Definitely can make the example more comprehensive, there is a cost on readability (of the example) though...
But really happy to add it if it is deemed useful.

**Comment by jmcazaux** [comment]
@mgu comments accounted for!

File: .pre-commit-hooks.yaml [hunk 3]
```diff
@@ -10,3 +10,9 @@
...
+  pass_filename: false
```
Comment (mgu) [inline .pre-commit-hooks.yaml]: Is it normal that there are no newline here ? (see the red circle)

Comment (jmcazaux) [reply .pre-commit-hooks.yaml]: I do think it matters in this context, but lets's go for a newline!

Comment (jmcazaux) [reply .pre-commit-hooks.yaml]: @mgu added new lines, thanks.

**Pull request merged**